### PR TITLE
Add osi-validation to CL2 and CL3

### PR DESCRIPTION
**Pull request by ClemensLinnhoff**
**Reference to a related issue in the repository**
#74 

**Add a description**

- [x] Add custom input and output rules
- [x] Add osi-validation trace file checks to CL2
- [x] Correct CL2 input trace files if error occur
- [x] Add osi-validation trace file checks to CL3
- [x] Correct CL3 input trace files if error occur
- [x] Correct model if errors occur
- [x] Change osi-validation to master branch

**Take this checklist as orientation for yourself, if this PR is ready for Maintainer Review**
- [x] My suggestion follows the [governance rules](https://openmsl.github.io/doc/OpenMSL/organization/governance_rules.html).
- [x] All commits of this PR are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits).
- [x] My changes generate no errors when passing CI tests. 
- [x] I updated all documentation (readmes incl. figures) according to my changes.
- [x] I have successfully implemented and tested my fix/feature locally.
- [x] Appropriate reviewer(s) are assigned.

### CODE REVIEW

**Comment by ClemensLinnhoff** [comment]
There seems to be a bug in OSI Validation, see https://github.com/OpenSimulationInterface/osi-validation/issues/78

**Comment by ClemensLinnhoff** [comment]
Waiting for https://github.com/OpenSimulationInterface/osi-validation/pull/79 to be merged.

**Comment by jdsika** [comment]
> Waiting for [OpenSimulationInterface/osi-validation#79](https://github.com/OpenSimulationInterface/osi-validation/pull/79) to be merged.

merged

File: .github/workflows/cl3.yml [hunk 1]
```diff
@@ -71,12 +71,12 @@ jobs:
           path: /tmp/esmini_fmu
           key: ${{ runner.os }}-esmini-fmu
 
-      - name: Cache OSI Field Checker
-        id: cache-osi-field-checker
-        uses: actions/cache@v4
+      - name: Cache osi-validation
+        id: cache-osi-validation
+        uses: actions/cache@v3
```
[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply .github/workflows/cl3.yml]: Good catch! This was a copy & paste error.

[the remark this answers is not in the document]
Comment (ClemensLinnhoff) [reply .github/workflows/cl3.yml]: I need to update this in the traffic participant model, too. I will do that once this PR is approved.

**Review by jdsika — APPROVED** [review]

**Pull request merged**